### PR TITLE
[dotnet] Enable runtime identifiers to be overriden at build time

### LINKED ISSUE #17707: Incorrect .app bundle when build with '-p:RuntimeIdentifier' option and <RuntimeIdentifiers> property
### Steps to Reproduce

1. Create maccatalyst project
2. Add 'RuntimeIdentifiers' property:
```xml
<RuntimeIdentifiers>maccatalyst-arm64;maccatalyst-x64</RuntimeIdentifiers>
```
3. Try to build the project with following command:
```
dotnet build -f:net7.0-maccatalyst -p:RuntimeIdentifier=maccatalyst-arm64
```
4. The application will be built, but it is empty (there are no dll and dylib libraries).
5. Run the application:
image

### Expected Behavior
I think the behavior should be like the `<TargetFrameworks>` option:

I have multiple frameworks in my project
```xml
<TargetFrameworks>net7.0-android;net7.0-ios;net7.0-maccatalyst</TargetFrameworks>
```
if I build the application using `dotnet build -p:TargetFramework=net7.0-android` then I will have only `net7.0-android` application assembled.

### Actual Behavior
incorrect (empty) app bundle

### Environment
dotnet 7.0.200
latest iOS workload
MacBook Pro (M1)

### Build Logs
[msbuild-binlog.txt](https://github.com/xamarin/xamarin-macios/files/10889321/msbuild-binlog.txt) - rename to .binlog

### PR DESCRIPTION
**Pull request by haritha-mohan**
Fixes #17707
The error target for when there is a conflict of interest in defining both the runtime identifier and runtime identifiers is called during the multi-rid builds, but not sure if the placement is the most ideal..

### CODE REVIEW

File: tests/dotnet/UnitTests/ProjectTest.cs [hunk 1]
```diff
@@ -431,6 +431,38 @@ public void IsNotMacBuild (ApplePlatform platform, string runtimeIdentifiers, bo
...
+		[TestCase (ApplePlatform.MacCatalyst, "maccatalyst-arm64;maccatalyst-x64")]
+		public void IsOverrideRuntimeIdentifier (ApplePlatform platform, string runtimeIdentifiers)
+		{
+			var project = "MySimpleApp";
+			Configuration.IgnoreIfIgnoredPlatform (platform);
+			Configuration.AssertRuntimeIdentifiersAvailable (platform, runtimeIdentifiers);
+			var project_path = GetProjectPath (project, runtimeIdentifiers: runtimeIdentifiers, platform: platform, out var appPath);
+			Clean (project_path);
+			var properties = GetDefaultProperties (runtimeIdentifiers);
+			properties ["RuntimeIdentifier"] = "maccatalyst-arm64";
```
Comment (rolfbjarne) [inline tests/dotnet/UnitTests/ProjectTest.cs:444]: Minor: building for x64 is faster than for arm64, so this speeds up the test a little bit:

```suggestion
			properties ["RuntimeIdentifier"] = "maccatalyst-x64";
```

File: tests/dotnet/UnitTests/ProjectTest.cs [hunk 2]
```diff
@@ -431,6 +431,38 @@ public void IsNotMacBuild (ApplePlatform platform, string runtimeIdentifiers, bo
...
+			DotNet.AssertBuild (project_path, properties);
```
Comment (rolfbjarne) [inline tests/dotnet/UnitTests/ProjectTest.cs:445]: You can get the warnings and assert that we get the expected one:

```suggestion
			var rv = DotNet.AssertBuild (project_path, properties);
			var warnings = BinLog.GetBuildLogWarnings (rv.BinLogPath).ToArray ();
			// the assert here
```

**Review by rolfbjarne — APPROVED** [review]
This looks great! Just a few minor comments.

**Review by rolfbjarne — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]
👍

**Review by tj-devel709 — APPROVED** [review]

**Pull request merged**